File: language/scala/spark/writeCsv.scala

pandaLovers.map(person => List(person.name, person.favoriteAnimal).toArray).mapPartitions { people =>
    val stringWriter = new StringWriter();
    val csvWriter = new CSVWriter(stringWriter); 
    csvWriter.writeAll(people.toList) 
    Iterator(stringWriter.toString)
}.saveAsTextFile(outFile)

File: language/py/spark/write_csv.py

outputFile = 'foo'

def writeRecords(records):
    
    """Write out CSV lines"""
    
    output = StringIO.StringIO()
    writer = csv.DictWriter(output, fieldnames=["name", "favoriteAnimal"]) 
    for record in records:
        writer.writerow(record) 
    return [output.getvalue()]
    
pandaLovers.mapPartitions(writeRecords).saveAsTextFile(outputFile)